src/main/java/ua/samoplavskaya/hw_patterns/Deep_copy/Person.java

```diff
@@ -0,0 +1,31 @@
+package ua.samoplavskaya.hw_patterns.Deep_copy;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Person implements Cloneable {
+    private String firstname;
+    private String lastname;
+    private int age;
+    private Address address;
+
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
```

Comment:
<img width="508" alt="image" src="https://github.com/ISamoplavskaya/alevel/assets/72043323/f07bc113-4e0a-4637-8447-163f6196fd58">

We can use more narrow type so as not to cast in code who calls clone()

N: In general, your implementation is okay, but there are some different ways:
https://chat.openai.com/share/430e9a44-92fa-41f4-bd7c-a86c8b6da252